src/pages/settings/Troubleshoot/TroubleshootPage.tsx

```diff
@@ -0,0 +1,178 @@
+import Section from '@components/Section';
+import TestToolMenu from '@components/TestToolMenu';
+import Text from '@components/Text';
+import TextLink from '@components/TextLink';
+import useEnvironment from '@hooks/useEnvironment';
+import useLocalize from '@hooks/useLocalize';
+import useTheme from '@hooks/useTheme';
+import useThemeStyles from '@hooks/useThemeStyles';
+import useWaitForNavigation from '@hooks/useWaitForNavigation';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import Navigation from '@libs/Navigation/Navigation';
+import * as App from '@userActions/App';
+import * as Report from '@userActions/Report';
+import type {TranslationPaths} from '@src/languages/types';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {OnyxKey} from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import SCREENS from '@src/SCREENS';
+
+const keysToPreserve: OnyxKey[] = [
+    ONYXKEYS.ACCOUNT,
+    ONYXKEYS.IS_CHECKING_PUBLIC_ROOM,
+    ONYXKEYS.IS_LOADING_APP,
+    ONYXKEYS.IS_SIDEBAR_LOADED,
+    ONYXKEYS.MODAL,
+    ONYXKEYS.NETWORK,
+    ONYXKEYS.SESSION,
+    ONYXKEYS.SHOULD_SHOW_COMPOSE_INPUT,
+    ONYXKEYS.NVP_TRY_FOCUS_MODE,
+    ONYXKEYS.PREFERRED_THEME,
+    ONYXKEYS.NVP_PREFERRED_LOCALE,
+    ONYXKEYS.CREDENTIALS,
+];
+
+type BaseMenuItem = {
+    translationKey: TranslationPaths;
+    icon: React.FC<SvgProps>;
+    action: () => void | Promise<void>;
+};
+
+type TroubleshootPageOnyxProps = {
+    shouldStoreLogs: OnyxEntry<boolean>;
+};
+
+type TroubleshootPageProps = TroubleshootPageOnyxProps;
+
+function TroubleshootPage({shouldStoreLogs}: TroubleshootPageProps) {
+    const {translate} = useLocalize();
+    const theme = useTheme();
+    const styles = useThemeStyles();
+    const {isProduction} = useEnvironment();
+    const [isConfirmationModalVisible, setIsConfirmationModalVisible] = useState(false);
+    const waitForNavigate = useWaitForNavigation();
+    const {isSmallScreenWidth} = useWindowDimensions();
+
+    const menuItems = useMemo(() => {
+        const debugConsoleItem: BaseMenuItem = {
+            translationKey: 'initialSettingsPage.troubleshoot.viewConsole',
+            icon: Expensicons.Gear,
+            action: waitForNavigate(() => Navigation.navigate(ROUTES.SETTINGS_CONSOLE)),
+        };
+
+        const baseMenuItems: BaseMenuItem[] = [
+            {
+                translationKey: 'initialSettingsPage.troubleshoot.clearCacheAndRestart',
+                icon: Expensicons.RotateLeft,
+                action: () => setIsConfirmationModalVisible(true),
+            },
+        ];
+
+        if (shouldStoreLogs) {
+            baseMenuItems.push(debugConsoleItem);
+        }
+
+        return baseMenuItems
+            .map((item) => ({
+                key: item.translationKey,
+                title: translate(item.translationKey),
+                icon: item.icon,
+                onPress: item.action,
+                containerStyle: [styles.pl10],
```

Comment:
I left this comment with a screenshot but just noting that this is giving us poor alignment.

Reply:
Nice, thank you! Going to run another test build too.